cumulusci/utils/fileutils.py

```diff
@@ -86,6 +88,69 @@ def load_from_source(
             yield path, f
 
 
+def proxy(funcname):
+    def func(self, *args, **kwargs):
+        real_func = getattr(self.fs, funcname)
+        return real_func(self.filename, *args, **kwargs)
+
+    func.__doc__ = getattr(base.FS, funcname).__doc__
+    return func
+
+
+class FSResource(str):
+    """Generalization of pathlib.Path to support S3, FTP, etc
```

Comment:
It makes me somewhat nervous to make this so flexible, because it makes it so much more important to ensure that the execution environment only has access to resources that the people with commit access to the project can be trusted with.

Reply:
I'm not sure it does make any security difference, really. If you have the auth information for an FTP site or an S3 bucket, you don't need this class to save your data there. And if you don't, it won't help you. (I have so far not done anything at all about providing access to S3 as a service)